web/hosts.go

```diff
@@ -127,35 +124,17 @@ func getTrentoAgentCheck(client consul.Client, node string) (*consulApi.HealthCh
 	return trentoAgentCheck, nil
 }
 
-func NewHostHandler(client consul.Client, subsService services.SubscriptionsService) gin.HandlerFunc {
+func NewHostHandler(hostsService services.HostsService, subsService services.SubscriptionsService) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		name := c.Param("name")
-		catalogNode, _, err := client.Catalog().Node(name, nil)
-		if err != nil {
-			_ = c.Error(err)
-			return
-		}
-
-		if catalogNode == nil {
-			_ = c.Error(NotFoundError("could not find host"))
-			return
-		}
 
-		trentoAgentCheck, err := getTrentoAgentCheck(client, name)
+		host, err := hostsService.GetByName(name)
 		if err != nil {
-			_ = c.Error(err)
-			return
-		}
-
-		systems, err := sapsystem.Load(client, name)
-		if err != nil {
-			_ = c.Error(err)
+			_ = c.Error(NotFoundError("could not find host"))
```

Comment:
The Hostname is not unique anymore, we need to change the service and the parameters (and the links around) basing everything on the ID